torch/_dynamo/symbolic_convert.py

```diff
@@ -319,6 +322,10 @@ def exit(self, tx, is_graph_break):
             return self.with_context.exit(tx)
 
 
+class SpeculationLogDivergence(Exception):
```

Comment:
I think you want this to inherit from AssertionFailure maybe?

Reply:
Done.